blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BlueOceanGraphListener.java

```diff
@@ -0,0 +1,21 @@
+package io.jenkins.blueocean.rest.impl.pipeline;
+
+import hudson.ExtensionPoint;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.input.InputStep;
+
+/**
+ *
+ * Listener for input step submission event
+ *
+ * @author Vivek Pandey
+ */
+public interface BlueOceanGraphListener extends ExtensionPoint{
+    /**
+     * This event is sent when an input step moves from paused to continue state. That is when an input form is submitted.
+     *
+      * @param inputStep {@link InputStep} which got executed by sumitting parameters
+     *  @param run {@link WorkflowRun} associated with this input step
+     */
+    void onStepContinue(InputStep inputStep, WorkflowRun run);
+}
```

Comment:
@vivek go nuts on it. Checkout the branch and do what you think is best. I am done now (just fixed some glitches in the matrix, ...karaoke I meant. ;)